### PR TITLE
Guesty new

### PR DESCRIPTION
**Pull request by liang611zh**
- add git ignore,
- modify url
- httpclient for encode url data format

### CODE REVIEW

File: src/ClientWrapper.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public function __construct($baseUrl, $logger=null)
      * @return array|mixed
      * @throws LauraException
      */
-    public function request($urlArray, $header, $data)
+    public function request($urlArray, $header, $data, $jsonEncode = true)
```
Comment (t3rryw1) [inline src/ClientWrapper.php:28]: Why do we need this flag?

File: src/HttpClient.php [hunk 2]
```diff
@@ -54,10 +55,15 @@ public function get($url, $headerArray)
      */
     public function post($url, $headerArray, $data)
     {
+        print_r($data);
```
Comment (t3rryw1) [inline src/HttpClient.php:69]: remove this

File: src/ClientWrapper.php (hunk 1 above)
Comment (liang611zh) [reply src/ClientWrapper.php:28]: because the auth data format is data-encode type

Comment (liang611zh) [reply src/ClientWrapper.php:28]: curl --location --request POST 'https://open-api.guesty.com/oauth2/token' \
--header 'accept: application/json' \
--data-urlencode 'grant_type=client_credentials' \
--data-urlencode 'scope=open-api' \
--data-urlencode 'client_secret={application client SECRET}'
--data-urlencode 'client_id={application client ID}' \

Comment (liang611zh) [reply src/ClientWrapper.php:28]: others request using

curl --request POST \
     --url https://open-api.guesty.com/v1/listings \
     --header 'Accept: application/json' \
     --header 'Content-Type: application/json' \
     --data '
{
     "title": "dddd",
     "origin": "bbb"
}
'

**Pull request merged**